tests/test_testclient.py

```diff
@@ -254,19 +254,25 @@ async def asgi(receive: Receive, send: Send) -> None:
 
 
 def test_websocket_not_block_on_close(test_client_factory: TestClientFactory) -> None:
+    cancelled = False
+
     def app(scope: Scope) -> ASGIInstance:
         async def asgi(receive: Receive, send: Send) -> None:
-            websocket = WebSocket(scope, receive=receive, send=send)
-            await websocket.accept()
-            while True:
-                await anyio.sleep(0.1)
+            nonlocal cancelled
+            try:
+                websocket = WebSocket(scope, receive=receive, send=send)
+                await websocket.accept()
+                await anyio.sleep_forever()
+            except anyio.get_cancelled_exc_class():
+                cancelled = True
+                raise
```

Comment:
Why is the cancelled exception here? Is this because this except was removed?

The addition of that `except` was on purpose, if I recall correctly.

Reply:
this is instead of websocket.should_close.is_set() no longer exists - so we need to find out that the async function ran and was cancelled

Reply:
did you mean to ask about this one: https://github.com/encode/starlette/pull/2814/files#diff-aca25e5f16c4fd49338ccdf3631f72455309335fee1e27f3d3b6016fa94ecedfL145 ?

this is because it's no longer possible to get a intentional cancelled_exc here because it will be caught by the CancelScope
if you do get an cancelled_exc here it's because someone issued a native asyncio cancel or manually raised a trio cancel, both of which should be propagated out of the websocket_connect cmgr

Reply:
Got it.

So the point is: since the WebSocket doesn't call `send` it can't receive a disconnect exception, and since it doesn't call `receive`, it doesn't receive a `websocket.disconnected` message. Then, the `TestClient` here will propagate the cancellation exception here, but the user shouldn't worry about it because the `TestClient` will catch it anyway.

Right?

Reply:
yeah, we use a CancelScope passed out with task_status.started(cs), this is then used to cancel the task on completion and collect a result. The CancelScope catches the cancellation that it causes to raise in the coro, unless there's another reason for cancellation which is then propagated, this is a fatal case and so the user should be notified

Reply:
So if there's a cancelled exception that is not from the TestClient, the user will see it, right?

I'm good with this.

Reply:
Yeah the only way is if the user was calling .cancel() directly or manually raising constructing a trio cancel (bypassing the private constructor protections)